compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_compatibility.rs

```diff
@@ -112,9 +112,29 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         // let _: &dyn Alias<Assoc = u32> = /* ... */;
         // ```
         let mut projection_bounds = FxIndexMap::default();
+        let mut auto_fill_bounds: SmallVec<[_; 4]> = smallvec![];
+        let mut push_auto_fill_receiver_bound = |pred: ty::Binder<'tcx, _>| {
```

Comment:
This is a bit overkill. Just take the projection and replace its def id with the one for `Receiver::Target`. 

At that point, this doesn't need to be a callback, just an additional call to `insert` into the projection bounds map that you can duplicate twice.